optaplanner-quarkus-integration/optaplanner-quarkus/deployment/src/main/java/org/optaplanner/quarkus/deployment/OptaPlannerProcessor.java

```diff
@@ -81,18 +86,37 @@ HotDeploymentWatchedFileBuildItem watchScoreDrl() {
     }
 
     @BuildStep
-    void registerAdditionalBeans(BuildProducer<AdditionalBeanBuildItem> additionalBeans) {
+    void registerAdditionalBeans(CombinedIndexBuildItem combinedIndex,
+            BuildProducer<AdditionalBeanBuildItem> additionalBeans) {
+        IndexView indexView = combinedIndex.getIndex();
+        if (hasNoOptaPlannerContent(indexView)) {
+            return;
+        }
         // The bean encapsulating the SolverFactory
         additionalBeans.produce(new AdditionalBeanBuildItem(OptaPlannerBeanProvider.class));
     }
 
+    private boolean hasNoOptaPlannerContent(IndexView indexView) {
+        // Only skip this extension if everything is missing. Otherwise, if some parts are missing, fail fast later.
+        return indexView.getAnnotations(DotNames.PLANNING_SOLUTION).isEmpty()
```

Comment:
Do we care about DRL (less than about easy score calculation)?

Reply:
We could do add a `classLoader.getResource(SolverBuildTimeConfig.DEFAULT_SCORE_DRL_URL) == null` check?

Reply:
That partially helps; now people might override the default location by solverConfig.xml. Where is the borderline of what else to check? What about the opposite approach: check just the annotations and provide a debug-level message explaining what's going on.
Because if we check the project just partially, for a couple of corner cases it will be very difficult for a user to investigate why the extension does just nothing.

Reply:
Very good point about the debug message!